### ISSUE TITLE
zero value for type="number" set to empty string on initial render

### ISSUE THREAD

**Issue by sentilesdal** [nkbt/react-debounce-input#62]
```
  getInitialState() {
    return {
      value: this.props.value || '';
    };
  },
```

should be

```
  getInitialState() {
    let value = this.props.value || '';
    
    if (this.props.type === 'number') {
      value = value || '0';
    }

    return {value};
  },
```

or some such equivalent.  thanks for the good work though!

**Comment by nkbt** [comment]
You're right! Thanks. I'll accept the PR if you are keen...

**Comment by burtyish** [comment]
I recently made the change below on our own fork.
We don't test for `type === 'number'` because we need the same functionality for `text` inputs as well as `number`.
@nkbt If you approve of my version, I don't mind making a PR.
```js
getInitialState() {
    return {
        value: this.props.value || this.props.value === 0 ? this.props.value : ''
    };
  },
```

**Comment by nkbt** [comment]
I think it may be better to do the other way around, plus cast to string like:

```js
  value: this.props.value === undefined ? '' : `${this.props.value}`
```

As far as I remember there is a code that relies on value being a string.

**Comment by alextrastero** [comment]
ping?

**Comment by nkbt** [comment]
Closed by #125
